pkg/registry/core/pod/strategy.go

```diff
@@ -205,6 +205,18 @@ type podEphemeralContainersStrategy struct {
 // EphemeralContainersStrategy wraps and exports the used podStrategy for the storage package.
 var EphemeralContainersStrategy = podEphemeralContainersStrategy{Strategy}
 
+func (podEphemeralContainersStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newPod := obj.(*api.Pod)
+	oldPod := old.(*api.Pod)
+
+	// Drop all changes except for pod.Spec.EphemeralContainers
```

Comment:
keep metadata.{name,namespace,resourceVersion,uid} as well. name/namespace/uid are immutable and ensure the API caller is referring to the same pod. resourceVersion ensures the caller is operating on a current version of the pod.